### PR TITLE
Add extend docs -1 

### PR DESCRIPTION
**Pull request by sherenewso2**

### CODE REVIEW

File: en/docs/extend/writing-a-custom-claim-handler.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+                return externalClaims;
+            }
+        }
+        ```
+
+        After you write the custom claim handler, and register the class as an OSGI bundle, you can deploy the OSGi bundle in WSO2 IS.
+
+---
+
+## Deploying the custom claim handler
```
Comment (gomathyk) [inline en/docs/extend/writing-a-custom-claim-handler.md:179]: Deploy? 

Please check other headings as well.

File: en/mkdocs.yml [hunk 2]
```diff
@@ -291,8 +291,22 @@ nav:
       - 'Configure Provisioning - To Remove':
         - 'Overview': guides/identity-provisioning/provisioning-overview.md
   - 'Extend':
-    - 'Authentication Script API': apis/authentication-script-reference.md
-    - 'Configuration Catalog': apis/configuration-catalog.md
+    - 'Authentication Script API': extend/authentication-script-reference.md
```
Comment (gomathyk) [inline en/mkdocs.yml:294]: Shall we change these headings with verb-like beginnings so that it is in sync with the guides and other docs?

**Review by gomathyk — CHANGES_REQUESTED** [review]

File: en/mkdocs.yml (hunk 2 above)
Comment (sherenewso2) [reply en/mkdocs.yml:294]: When referring to other apis in the "APIs" section, we have not used a verb in the beginning. We could change them all to "Use xxxx API" but I personally think this format is fine for APIs. WDYT?

File: en/docs/apis/service-provider-configurations-used-with-apis.md [hunk 3]
```diff
@@ -145,7 +145,7 @@ This section guides you through the configurations you can include in a service
...
                 <li><b>idPInitSSOEnabled</b> <code>[Boolean]</code> <br>This is set <code>true</code> to enable IdP initiated SSO profile for the service provider.</li>
                 <li><b>idPInitSLOEnabled</b> <code>[Boolean]</code> <br>This is set <code>true</code> to enable IdP initiated SLO profile for the service provider.</li>
-                <li><b>assertionQueryRequestProfileEnabled</b> <code>[Boolean]</code> <br>This is set <code>true</code> to enable Assertion Query Request Profile to query assertions that are persisted to the database when you log in to the service provider application. For more information, see <a href="../../learn/querying-saml-assertions">Querying SAML Assertions</a>.</li>           
+                <li><b>assertionQueryRequestProfileEnabled</b> <code>[Boolean]</code> <br>This is set <code>true</code> to enable Assertion Query Request Profile to query assertions that are persisted to the database when you log in to the service provider application. For more information, see <a href="../../guides/login/query-saml-assertions/">Querying SAML Assertions</a>.</li>
```
Comment (gomathyk) [inline en/docs/apis/service-provider-configurations-used-with-apis.md:148]:
```suggestion
                <li><b>assertionQueryRequestProfileEnabled</b> <code>[Boolean]</code> <br>This is set <code>true</code> to enable Assertion Query Request Profile to query assertions that are persisted to the database when you log in to the service provider application. For more information, see <a href="../../../guides/login/query-saml-assertions/">Querying SAML Assertions</a>.</li>           
```

File: en/docs/developer-portal/applications/oidc-configurations.md [hunk 4]
```diff
@@ -14,8 +14,8 @@ Now that you have created an application in the Identity Server using a template
 
 This certificate is used to validate the signatures of the signed requests from the application (service provider) to WSO2 IS. Therefore, the certificate is used for the following scenarios:
 
-- [Configuring single sign-on](../../learn/configuring-single-sign-on-saml#setting-up) - the certificate is used to validate the signature of the SAML2 authentication requests and the SAML2 logout requests that are sent by the service provider.
-- [Passing OIDC authentication request parameters](../../learn/passing-oidc-authentication-request-parameters-in-a-request-object) - the certificate is used to:
+- [Configuring single sign-on](../../guides/login/saml-app-config-advanced/#certificate-alias) - the certificate is used to validate the signature of the SAML2 authentication requests and the SAML2 logout requests that are sent by the service provider.
```
Comment (gomathyk) [inline en/docs/developer-portal/applications/oidc-configurations.md:17]:
```suggestion
- [Configuring single sign-on](../../../guides/login/saml-app-config-advanced/#certificate-alias) - the certificate is used to validate the signature of the SAML2 authentication requests and the SAML2 logout requests that are sent by the service provider.
```

File: en/docs/developer-portal/applications/oidc-configurations.md [hunk 5]
```diff
@@ -14,8 +14,8 @@ Now that you have created an application in the Identity Server using a template
...
+- [Passing OIDC authentication request parameters](../../guides/login/oidc-request-object) - the certificate is used to:
```
Comment (gomathyk) [inline en/docs/developer-portal/applications/oidc-configurations.md:18]:
```suggestion
- [Passing OIDC authentication request parameters](../../../guides/login/oidc-request-object) - the certificate is used to:
```

File: en/docs/extend/provisioning/extend-scim2-user-schemas.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
+# Extend SCIM 2.0 User Schemas
+ 
+The System for Cross-domain Identity Management (SCIM) is a specification that is designed to manage user identities in cloud-based applications and services in a standardized way to enable interoperability, security, and scalability. It is an emerging open standard which provides RESTful APIs for easier, cheaper, and faster way for creating, provisioning, and maintaining identities. The latest version SCIM 2.0 was released as IETF RFC in September 2015.
+
+!!! Note 
+    SCIM 2.0 is supported by default in WSO2 Identity Server version 5.4.0 onwards. If you are using WSO2 Identity Server 5.4.0 or a later version, see [SCIM 2.0 REST APIs](../develop/scim2-rest-apis.md) for instructions on how to use SCIM 2.0 OOTB.
```
Comment (gomathyk) [inline en/docs/extend/provisioning/extend-scim2-user-schemas.md:6]:
```suggestion
    SCIM 2.0 is supported by default in WSO2 Identity Server version 5.4.0 onwards. If you are using WSO2 Identity Server 5.4.0 or a later version, see [SCIM 2.0 REST APIs](../../apis/scim2-rest-apis.md) for instructions on how to use SCIM 2.0 OOTB.
```

File: en/docs/extend/provisioning/extensible-scim-user-schemas-with-wso2-identity-server.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
+# Extensible SCIM User Schemas With WSO2 Identity Server
```
Comment (gomathyk) [inline en/docs/extend/provisioning/extensible-scim-user-schemas-with-wso2-identity-server.md:1]:
```suggestion
# Extensible SCIM User Schemas with WSO2 Identity Server
```

File: en/docs/extend/user-mgt/user-management-errors-event-listener.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
+# User Management Errors Event Listener
+
+The `UserManagementErrorEventListener` (i.e. `org.wso2.carbon.user.core.listener.UserManagementErrorEventListener`) is a new type of event listener that facilitates additional activities in the event of failures when executing user management scenarios or operations. The relevant caller class for this listener is `org.wso2.carbon.user.core.common.AbstractUserManagementErrorListener`. This type of event listener is similar to the [UserOperation Listeners](../../develop/user-store-listeners). The registered event listeners of this type are called when there is a failure while doing user management related tasks.
```
Comment (gomathyk) [inline en/docs/extend/user-mgt/user-management-errors-event-listener.md:3]:
```suggestion
The `UserManagementErrorEventListener` (i.e. `org.wso2.carbon.user.core.listener.UserManagementErrorEventListener`) is a new type of event listener that facilitates additional activities in the event of failures when executing user management scenarios or operations. The relevant caller class for this listener is `org.wso2.carbon.user.core.common.AbstractUserManagementErrorListener`. This type of event listener is similar to the [UserOperation Listeners](../../../extend/user-mgt/user-store-listeners). The registered event listeners of this type are called when there is a failure while doing user management related tasks.
```

File: en/docs/extend/user-mgt/user-store-listeners.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+!!! tip
+    Make note of the following.
+    
+    1.  The `getExecutionOrderId()` method can return any random value. This is important when there is more than one listener in the Identity Server and you need to consider the execution order of them.
+
+    2.  All the methods return a ` boolean` value. This value is mentioned regardless of whether you want to execute the next listener or not.
+    
+The following are the steps to configure the custom implementation.
+
+1.  Listeners are registered as OSGI components. Therefore you need to register this class in an OSGI framework. You can go through **[this](https://svn.wso2.org/repos/wso2/people/asela/user-mgt/custom-listener/),** which is the complete sample project.
```
Comment (gomathyk) [inline en/docs/extend/user-mgt/user-store-listeners.md:96]:
```suggestion
1.  Listeners are registered as OSGI components. Therefore you need to register this class in an OSGI framework. You can go through [this](https://svn.wso2.org/repos/wso2/people/asela/user-mgt/custom-listener/),which is the complete sample project.
```

File: en/docs/extend/user-mgt/write-a-custom-claim-handler.md [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+
+## Deploy the custom claim handler
+
+Follow the steps below to deploy the custom claim handler that you implemented:
+
+1.  Add the compiled OSGi bundle to the `IS_HOME/repository/components/dropins` directory.
+
+2.  Open the ` deployment.toml ` file in ` IS_HOME/repository/conf` folder and add the custom claim handler package name under the `[authentication.framework.extensions]` tag as shown below.   
+    
+    ```toml
```
Comment (gomathyk) [inline en/docs/extend/user-mgt/write-a-custom-claim-handler.md:187]: Please check the alignment

File: en/docs/extend/user-mgt/write-a-custom-event-handler.md [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+- `PRE_ADD_USER`
+- `POST_ADD_USER`
+
+!!! info
+    The other events available with WSO2 Identity Server can be found from the `Event` class [here](https://github.com/wso2/carbon-identity-framework/blob/master/components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/IdentityEventConstants.java)
+
+
+---
+
+## Writing an event handler
```
Comment (gomathyk) [inline en/docs/extend/user-mgt/write-a-custom-event-handler.md:38]:
```suggestion
## Write an event handler
```

File: en/docs/quick-starts/oidc-session-management.md [hunk 12]
```diff
@@ -122,7 +122,7 @@ If you have your own application, click the button below.
     particular page after the RP sends the OpenID Connect logout request.
     For more information on how to configure the redirect logout, see
     [OpenID Connect Logout URL
-    Redirection](../../learn/openid-connect-logout-url-redirection/).
+    Redirection](../../guides/login/oidc-logout-url-redirection/).
```
Comment (gomathyk) [inline en/docs/quick-starts/oidc-session-management.md:125]:
```suggestion
    Redirection](../../../guides/login/oidc-logout-url-redirection/).
```

**Review by gomathyk — CHANGES_REQUESTED** [review]

**Review by gomathyk — APPROVED** [review]

**Pull request merged**